bundles/org.pitest.pitclipse.launch.ui/src/org/pitest/pitclipse/launch/ui/PitMutatorsTab.java

```diff
@@ -0,0 +1,366 @@
+ * use this file except in compliance with the License.  You may obtain a copy
+ * of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+
+package org.pitest.pitclipse.launch.ui;
+
+import static org.pitest.pitclipse.core.preferences.PitPreferences.INDIVIDUAL_MUTATORS;
+import static org.pitest.pitclipse.core.preferences.PitPreferences.MUTATORS;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.viewers.ArrayContentProvider;
+import org.eclipse.jface.viewers.CellLabelProvider;
+import org.eclipse.jface.viewers.CheckboxTableViewer;
+import org.eclipse.jface.viewers.TableViewerColumn;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.jface.viewers.ViewerCell;
+import org.eclipse.jface.viewers.ViewerFilter;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.program.Program;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.pitest.mutationtest.engine.gregor.config.Mutator;
+import org.pitest.pitclipse.core.Mutators;
+import org.pitest.pitclipse.core.PitCoreActivator;
+import org.pitest.pitclipse.runner.config.PitConfiguration;
+import org.pitest.pitclipse.ui.utils.PitclipseUiUtils;
+
+/**
+ * Tab allowing to configure a PIT analyze.
+ * @author Jonas Kutscha
+ */
+public final class PitMutatorsTab extends AbstractLaunchConfigurationTab {
+    public static final String NAME = "Mutators";
+
+    private static final int NUMBER_OF_COLUMNS = Mutators.getMainGroup().size() + 2;
+    private static final String DESCRIPTION_TEXT = "Select the mutators used to alter the code.";
+    private static final String MUTATOR_LINK_TEXT = "See the documentation on Pitest.org";
+    private static final String MUTATOR_LINK = "https://pitest.org/quickstart/mutators/";
+    public static final String CUSTOM_MUTATOR_RADIO_TEXT = "Mutators selected below";
+    private static final String CUSTOM_MUTATOR_RADIO_DATA = "CUSTOM";
```

Comment:
It looks like this constant is unused (it's used to set the data of a button but this data doesn't seem to have any purpose). Could you either remove it or add a comment explaining why it's needed?

Reply:
Has now a comment

Reply:
Thanks, I now understand what's going on. However I believe that it's not clear enough: in order to understand, a reader has to:

1. Read the constant's comment
2. Go to the class
3. Dig into `widgetSelection`'s implementation
4. Check how buttons are created and notice that their data is the name of  mutator group (`button.setData(mutatorGroup.name());`

That's too much. Keep in mind that the reader may not be familiar with the code and that, ideally, the code should explain itself without requiring the reader to dig. In this regards I'd suggest the following changes (tentative wording, you can rephrase it if you wish but please keep the same information):

```java
    /**
     * The data of the "Custom Mutators" button. Each button requires a data that uniquely identifies it
     * so that {@link UpdateDialogOnCurrentMutatorGroupChanged#widgetSelected} can determine
     * when a new mutator group is selected.
     * 
     * Other buttons do not require a constant because their data is set to the name of the corresponding
     * PIT mutator group.
     */
    private static final String CUSTOM_MUTATOR_RADIO_DATA = "CUSTOM";

    [...]

    /**
     * Called each time a mutator group is selected. If this mutator group was not previously
     * selected, the launch configuration dialog is updated and the table may be disable if unused.
     */
    private class UpdateDialogOnCurrentMutatorGroupChanged extends SelectionAdapter {
        @Override
        public void widgetSelected(SelectionEvent event) {
            final String oldMutatorGroup = currentMutatorGroup;
            currentMutatorGroup = (String) event.widget.getData();
            if (((Button) event.widget).getSelection() && !oldMutatorGroup .equals(currentMutatorGroup )) {
                updateLaunchConfigurationDialog();
                disableTableIfUnused();
            }
        }
    }
```
Notice how I also renamed the `ButtonSelectionListener` class as well as some variables to make their purpose more explicit.

Reply:
For the record: the constant does not exist anymore and isn't needed because `CUSTOM` is now a value of the `Mutators` enum